src/mongo/nodes.ts

```diff
@@ -74,13 +54,18 @@ export class MongoServerNode implements IMongoServer {
 		return MongoServerNode.getMongoDatabaseNodes(this._connectionString, this);
 	}
 
-	static async getMongoDatabaseNodes(connectionString: string, parentNode: IMongoServer): Promise<MongoDatabaseNode[]> {
-		const db = await MongoClient.connect(connectionString);
+	static async getMongoDatabaseNodes(connectionString: string, parentNode: IMongoServer): Promise<INode[]> {
+		let db: Db;
 		try {
+			db = await MongoClient.connect(connectionString);
 			const value: { databases: { name }[] } = await db.admin().listDatabases();
 			return value.databases.map(database => new MongoDatabaseNode(database.name, parentNode));
+		} catch (error) {
+			return [new ErrorNode(error.message)];
```

Comment:
Can the message property be a very long string? Might consider limiting the length of the text by trimming it.

Reply:
It can be a long string, but you can always expand the explorer view to see more of the message. Ideally we would be able to wrap the message, but I don't think that's possible.